### PR TITLE
ipatests: Add integration tests for External IdP support

### PR DESCRIPTION
**Pull request by amore17**
Tests for [RFE]: Added integration tests for external IdP
authentication with keycloak-17 as identity provider.

Related : https://pagure.io/freeipa/issue/8805
Related: https://pagure.io/freeipa/issue/8803
Related: https://pagure.io/freeipa/issue/8804

### CODE REVIEW

File: ipatests/pytest_ipa/integration/create_quarkus.py [hunk 1]
```diff
@@ -0,0 +1,153 @@
...
+    ExecStart=/opt/keycloak/bin/kc.sh start
+    TimeoutStartSec=600
+    TimeoutStopSec=600
+
+    [Install]
+    WantedBy=multi-user.target
+    """)
+    host.put_file_contents("/etc/systemd/system/keycloak.service", contents)
+    host.run_command(["systemctl", "daemon-reload"])
+    host.run_command(["setenforce", "0"])
```
Comment (abbra) [inline ipatests/pytest_ipa/integration/create_quarkus.py:79]: In my tests this was not needed, as long as you relabel content in /opt/keycloak with `restorecon -R /opt/keycloak'.

**Review by abbra — CHANGES_REQUESTED** [review]
Thank you for the PR! I have only one change requested -- remove permissive mode, it is not really required.

**Review by mrizwan93 — CHANGES_REQUESTED** [review]
Hello @amore17 
Please fine comments inline.

File: ipatests/pytest_ipa/integration/create_quarkus.py [hunk 2]
```diff
@@ -0,0 +1,152 @@
...
+    host.run_command(["chown", "-R", "keycloak:", dir])
+    host.run_command(["chmod", "o+x", "{0}/bin/".format(dir)])
+    host.run_command(["restorecon", "-R", dir])
+
+    # setup TLS certificate using IPA CA
+    host.run_command(["kinit", "-k"])
+    host.run_command(["ipa", "service-add", "HTTP/{0}".format(host.hostname)])
+    key = "/etc/pki/tls/private/keycloak.key"
+    crt = "/etc/pki/tls/certs/keycloak.crt"
+    keystore = "/etc/pki/tls/private/keycloak.store"
```
[the remark this answers is not in the document]
Comment (abbra) [reply ipatests/pytest_ipa/integration/create_quarkus.py:33]: Why? These do not belong to the runtime, they are part of the test environment.

[the remark this answers is not in the document]
Comment (amore17) [reply ipatests/pytest_ipa/integration/create_quarkus.py]: Yes they are related to test environment.

**Comment by amore17** [comment]
Tested also in idm-ci environment using @freeipa/freeipa-master-nightly

**Comment by amore17** [comment]
[logs](http://freeipa-org-pr-ci.s3-website.eu-central-1.amazonaws.com/jobs/86ba931c-d758-11ec-9976-fa163e0ccb17)

**Comment by abbra** [comment]
LGTM. I'll add ACK after the remaning two tests complete.

**Comment by abbra** [comment]
master:

* 5ca4e8ee33cb2de7f0e05b91dfe9cc41829609cd pr-ci definitions: add external idp related jobs.
* 9cc703fd0ed1afc24dc865ed967264c7f7340ac7 ipatests: Add integration tests for External IdP support

**Comment by abbra** [comment]
@amore17 please submit a manual backport, automation failed to create a clean merge to ipa-4-9.

**Pull request closed without merging**